src/procedures/__tests__/SetController.ts

```diff
@@ -0,0 +1,181 @@
+    );
+
+    tokenFactoryMockStub = tokenFactoryMock.mock(
+      'getSecurityTokenInstanceFromTicker',
+      securityTokenMock.getMockInstance()
+    );
+
+    contextMock.set('contractWrappers', wrappersMock.getMockInstance());
+    wrappersMock.set('tokenFactory', tokenFactoryMock.getMockInstance());
+  });
+
+  afterEach(() => {
+    restore();
+  });
+
+  describe('Types', () => {
+    test('should extend procedure and have SetController type', async () => {
+      target = new SetController(
+        {
+          symbol: params1.symbol,
+          controller: params1.controller,
+        },
+        contextMock.getMockInstance()
+      );
+      expect(target instanceof Procedure).toBe(true);
+      expect(target.type).toBe('SetController');
+    });
+  });
+
+  describe('SetController', () => {
+    test('should throw if there is no valid security token being provided', async () => {
+      // Instantiate SetController with incorrect security symbol
+      target = new SetController(
+        {
+          symbol: params1.symbol,
+          controller: params1.controller,
+        },
+        contextMock.getMockInstance()
+      );
+
+      tokenFactoryMock.set(
+        'getSecurityTokenInstanceFromTicker',
+        stub()
+          .withArgs({ address: params1.symbol })
+          .throws()
+      );
+
+      expect(target.prepareTransactions()).rejects.toThrow(
+        new PolymathError({
+          code: ErrorCode.ProcedureValidationError,
+          message: `There is no Security Token with symbol ${params1.symbol}`,
+        })
+      );
+    });
+
+    test('should call error on inappropriate params address', async () => {
+      // Instantiate SetController with incorrect args instead
+      target = new SetController(
+        {
+          symbol: params1.symbol,
+          controller: 'Inappropriate',
+        },
+        contextMock.getMockInstance()
+      );
+
+      // Real call rejects
+      expect(target.prepareTransactions()).rejects.toThrowError(
+        new PolymathError({
+          code: ErrorCode.ProcedureValidationError,
+          message: `Controller address "Inappropriate" is invalid.`,
+        })
+      );
+    });
+
+    test('should throw if account address is different than owner address', async () => {
+      securityTokenMock.mock('owner', Promise.resolve('0x01'));
+      contextMock.set(
+        'currentWallet',
+        new Wallet({ address: () => Promise.resolve('0x02') })
+      );
+
+      // Instantiate SetController
+      target = new SetController(
+        {
+          symbol: params1.symbol,
+          controller: params1.controller,
+        },
+        contextMock.getMockInstance()
+      );
+
+      // Real call rejects
+      expect(target.prepareTransactions()).rejects.toThrowError(
+        new PolymathError({
+          code: ErrorCode.ProcedureValidationError,
+          message: `You must be the owner of this Security Token to set the controller`,
+        })
+      );
+    });
+
```

Comment:
Description should be rewritten like:
”should add a transaction to the queue to set a controller on the security token”